Modules/SubModule_Population/xASL_adm_GzipAllFiles.m

```diff
@@ -39,47 +43,65 @@ function xASL_adm_GzipAllFiles(ROOT, bFolder, bUseLinux)
         PathToSearch = xASL_adm_UnixPath(ROOT);
         system(['for i in `find ' PathToSearch '/* | grep -E \.nii$`; do gzip -1 -f -q -v "$i"; done'], '-echo');
         return;
+    else
+        %% 2) Otherwise use the multithreaded SuperGzip for Windows
+        PathList = xASL_adm_GetFileList(ROOT, '^.*\.(nii)$', 'FPListRec', [0 Inf], false);
+        fprintf('\n%s\n',['G-zZzZipping ' num2str(length(PathList)) ' files']);
+        n_THREADS = min([str2num(getenv('NUMBER_OF_PROCESSORS')) length(PathList)]);
+        if ~isempty(EXTERNAL)
+            PathToSuperGzip = fullfile(EXTERNAL, 'SuperGZip', 'SuperGZip_Windows.exe');
+            command = [PathToSuperGzip ' -p 0 -n ' num2str(n_THREADS) ' -v 1 ' ROOT ' *.nii'];
+            exit_code = system(command);
+        else
+            exit_code = 1;
+        end
+        if exit_code == 0
+            fprintf('Gzipping Niftis Successful...\n')
+        else
+            warning('An error occurred while using SuperGzip')
+        end
     end
 
     %% ----------------------------------------------------
     %% 2) Slower Matlab version
     if bFolder
         PathList = xASL_adm_GetFileList(ROOT, '^.*$', 'FPList', [0 Inf], true);
         fprintf('\n%s\n',['G-zZzZipping ' num2str(length(PathList)) ' folders']);
-    else
-        PathList = xASL_adm_GetFileList(ROOT, '^.*\.(nii|wav|bmp)$', 'FPListRec', [0 Inf], false);
-        fprintf('\n%s\n',['G-zZzZipping ' num2str(length(PathList)) ' files']);
-    end
-    
-    fprintf('\n%s\n',['G-zZzZipping ' num2str(length(PathList)) ' files']);
-    fprintf('Progress:   ');
-
-    for iList=1:length(PathList)
-        xASL_TrackProgress(iList,length(PathList));
+%     else
+%         PathList = xASL_adm_GetFileList(ROOT, '^.*\.(nii|wav|bmp)$', 'FPListRec', [0 Inf], false);
+%         fprintf('\n%s\n',['G-zZzZipping ' num2str(length(PathList)) ' files']);
         
-        try
-            if bFolder
-               zip([PathList{iList} '.zip'], PathList{iList});
-               xASL_adm_DeleteFileList(PathList{iList}, '^.*$', true, [0 Inf]);
-               xASL_delete(PathList{iList});
-            else           
-
-                Fname = PathList{iList};
-                [~, ~, Fext] = xASL_fileparts(Fname);
-                if strcmp(Fext,'.nii') || strcmp(Fext,'.wav') || strcmp(Fext,'.bmp')
-                    if exist([Fname '.gz'],'file')
-                        delete([Fname '.gz']); 
-                    end
-                    gzip(Fname);
-                    delete(Fname);
-                end
-            end
-        catch ME
-            warning('Something went wrong, continuing with next one');
-            fprintf('%s\n',ME.message);
-        end
     end
-    fprintf('\n');
+    
```

Comment:
why is the slower version commented out? Should still be available in case the faster Linux or Windows versions crash. I would add a try-catch statement that runs this slow part if the zipping crashed, with feedback warning why it crashed (`catch ME; fprintf(ME.message)`).

Reply:
Reverted 👍